llvm/lib/Target/SPIRV/SPIRVUtils.h

```diff
@@ -88,6 +88,10 @@ Type *getMDOperandAsType(const MDNode *N, unsigned I);
 // name, otherwise return an empty string.
 std::string getOclOrSpirvBuiltinDemangledName(StringRef Name);
 
+// If Type is a pointer type and it is not opaque pointer, return its
+// element type, otherwise return Type.
+const Type *maybeGetPointerEltType(const Type *Type);
```

Comment:
Don't like 'maybe' here, something like 'getTypedPtrEltType' would be better IMO. BTW are you sure there's no existing function for that?

Reply:
Ok.
> BTW are you sure there's no existing function for that?
I have not found such an existing function.